MetaMask/metamask-extension
# docs: Privacy settings manual scenarios

## **Description**

This PR documents the manual test scenarios for several privacy settings functionalities. Below is a summary of the validations included in these manual scenarios:

1. Privacy settings: Check “network details check” toggle works correctly

- Validate that the 'Network details check' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, MetaMask checks the network details against the information provided by chainid.network when attempting to connect to a blockchain network.

2. Privacy settings: Check “IPFS gateway” toggle works correctly

- Validate that the 'IPFS gateway' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, MetaMask shows images of NFTs.
- Validate that when the toggle is turned off, NFT images are replaced by placeholders and I am prompted to turn on IPFS resolution.

3. Privacy settings: Check “Show ENS domains in address bar” toggle works correctly

- Validate that the 'Show ENS domains in address bar' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, ENS domain name is displayed alongside the public address.

4. Privacy settings: Check “Display NFT Media” toggle works correctly

- Validate that the 'Display NFT media' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, MetaMask displays NFT media and data.
- Validate that NFT autodetection relies on this setting and won't be available when this is turned off.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26764?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/21963, https://github.com/MetaMask/metamask-extension/issues/21964, https://github.com/MetaMask/metamask-extension/issues/21966, https://github.com/MetaMask/metamask-extension/issues/21967

## **Manual testing steps**

Please click link for the [rendered view](https://github.com/MetaMask/metamask-extension/tree/privacy-settings-manual-scenarios/test/manual-scenarios/privacy-settings)

--> test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md
https://github.com/user-attachments/assets/9a60b564-7503-44b9-9ca0-810e857a4529

--> test/manual-scenarios/privacy-settings/check "IPFS gateway" toggle works correctly.md
https://github.com/user-attachments/assets/3e2e5135-9343-4d8c-808f-e3f599928d52

--> test/manual-scenarios/privacy-settings/check "Show ENS domains in address bar" toggle works correctly.md
https://github.com/user-attachments/assets/79df9cc4-0083-458f-be06-d524c7ca135f

--> test/manual-scenarios/privacy-settings/check "Display NFT media" toggle works correctly.md
https://github.com/user-attachments/assets/a54ef74f-00b4-4554-a7ac-9cd1e53800a4
